Update Targeted Messaging Outreach

hectorgomezv opened the pull request:
## Summary
This PR adds `sourceId`, `type`, `teamName`, `sourceFile`, and `sourceFileProcesedDate` to the Targeted Messaging `Outreach` entity.

## Changes
- Adds a database migration to add the fields.
- Adjusts tests and `TargetedMessagingDatasource.createOutreach`.

iamacook commented on src/datasources/targeted-messaging/entities/outreach.entity.ts:
```diff
@@ -3,6 +3,11 @@ export class Outreach {
+  source_id: number;
+  type: string;
+  team_name: string;
+  source_file: string;
+  source_file_processed_date: string;
```
Both `source_file` and `source_file_processed_date` can be `null`. I'm not 100% sure, but as it's a `TIMESTAMP`, shouldn't `source_file_processed_date` be a `Date` if defined instead of a `string`?

hectorgomezv replied:
I initially thought we would need to thread these values as strings when dealing directly with the `postgres` library, but you are right, it seems we can use `Date` directly so need to parse them twice. I've changed it in [bfb327e](https://github.com/safe-global/safe-client-gateway/pull/2010/commits/bfb327e54f8e3c1713ed4922a6f61d1ed48151c2)

This can also be applied to the other DB-related entities like `DbTargetedSafe` or `DbSubmission`. Will change those in a separate PR as they are not related to this one.

iamacook replied:
We should definitely check/adjust for the `Date` coercion accordingly when we move this to the ORM as that uses `pg` instead of `postgres`, cc @PooyaRaki

iamacook commented on src/datasources/targeted-messaging/targeted-messaging.datasource.ts:
```diff
@@ -60,6 +69,13 @@ export class TargetedMessagingDatasource
+      teamName: outreach.team_name,
+      sourceFile: outreach.source_file,
+      sourceFileProcessedDate: outreach.source_file_processed_date
+        ? new Date(outreach.source_file_processed_date)
```
Follow on from my type-related comment. Aren't these already `Date`s? (Would stand for `startDate`, `endDate`, `created_at` and `updated_at` as well.

hectorgomezv replied:
Absolutely! Answered in https://github.com/safe-global/safe-client-gateway/pull/2010#discussion_r1803861284

iamacook commented on src/domain/targeted-messaging/entities/create-outreach.dto.entity.ts:
```diff
@@ -1,9 +1,19 @@
+  sourceFileProcessedDate: z.coerce.date().nullish().default(null),
 });
 
 export class CreateOutreachDto
```
```suggestion
export class CreateOutreachDto implements z.infer<CreateOutreachDtoSchema>
```

hectorgomezv replied:
It would be `implements z.infer<typeof CreateOutreachDtoSchema>`, right? That's on the next line (20) (https://github.com/safe-global/safe-client-gateway/blob/eb7c106d7fd453178e976bf756c54c4580cb51e5/src/domain/targeted-messaging/entities/create-outreach.dto.entity.ts#L20)

Maybe I'm not understanding your comment, sorry.

iamacook replied:
You're right. My preview window didn't show the next line 🥺

hectorgomezv commented:
I've added `sourceFileChecksum` to the `Outreach` entity as the file integrity should be checked before parsing.

I've updated the PR description accordingly.
